### ISSUE TITLE
saving_utils.compile_args_from_training_config does not work with multiple loss fuctions

### ISSUE THREAD

**Issue by 7Z0nE** [tensorflow/tensorflow#36492]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): YES
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NO
- TensorFlow installed from (source or binary):binary
- TensorFlow version (use command below):TF VERSION:  1.15.0-rc2
- Python version: 3.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Describe the current behavior**
Error when loading a saved keras.model that was compiled with a list of loss functions.
### Stacktrace
```
Traceback (most recent call last):
  File "reproduce_issue_multiloss.py", line 27, in <module>
    model = keras.models.load_model("model.hdf5")
  File "/home/X/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/keras/saving/save.py", line 143, in load_model
    return hdf5_format.load_model_from_hdf5(filepath, custom_objects, compile)
  File "/home/X/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/keras/saving/hdf5_format.py", line 178, in load_model_from_hdf5
    training_config, custom_objects))
  File "/home/X/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/training/tracking/base.py", line 457, in _method_wrapper
    result = method(self, *args, **kwargs)
  File "/home/X/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/training.py", line 336, in compile
    self.loss, self.output_names)
  File "/home/X/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/training_utils.py", line 1355, in prepare_loss_functions
    loss_functions = nest.map_structure(get_loss_function, loss)
  File "/home/X/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/util/nest.py", line 536, in map_structure
    structure[0], [func(*x) for x in entries],
  File "/home/X/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/util/nest.py", line 536, in <listcomp>
    structure[0], [func(*x) for x in entries],
  File "/home/X/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/training_utils.py", line 1092, in get_loss_function
    name=loss_fn.__name__,
AttributeError: 'MeanSquaredError' object has no attribute '__name__'
```

**Describe the expected behavior**
The model is loaded and compiled without error.

**Code to reproduce the issue**
```
import tensorflow as tf
import tensorflow.keras as keras
import numpy as np

print("TF VERSION: ", tf.__version__)

inputs = keras.Input(2)
d1 = keras.layers.Dense(4)
d2 = keras.layers.Dense(4)
o1 = d1(inputs)
o2 = d2(inputs)

# make a model with multiple outputs
model = keras.Model(inputs=inputs, outputs=[o1, o2])

# compile the model with multiple losses
model.compile(loss=[keras.losses.MeanSquaredError(), keras.losses.MeanSquaredError()])

# try to feed a batch through the model
batch = np.linspace(0, 9, 10).reshape(5, 2)
outs = model.predict(batch)
print(outs)

# save and load the model
model.save("model.hdf5")

model = keras.models.load_model("model.hdf5")

```

**Other info / logs**
The error is related to the code in function ` compile_args_from_training_config` from `/python/keras/saving/saving_utils.py`

```
def compile_args_from_training_config(training_config, custom_objects=None):
  """Return model.compile arguments from training config."""
  if custom_objects is None:
    custom_objects = {}

  optimizer_config = training_config['optimizer_config']
  optimizer = optimizers.deserialize(
      optimizer_config, custom_objects=custom_objects)

  # Recover loss functions and metrics.
  loss_config = training_config['loss']  # Deserialize loss class.
  if isinstance(loss_config, dict) and 'class_name' in loss_config:
    loss_config = losses.get(loss_config)
  loss = nest.map_structure(
      lambda obj: custom_objects.get(obj, obj), loss_config)
  metrics = nest.map_structure(
      lambda obj: custom_objects.get(obj, obj), training_config['metrics'])
  weighted_metrics = nest.map_structure(
      lambda obj: custom_objects.get(obj, obj),
      training_config.get('weighted_metrics', None))
  sample_weight_mode = training_config['sample_weight_mode']
  loss_weights = training_config['loss_weights']

  return dict(
      optimizer=optimizer,
      loss=loss,
      metrics=metrics,
      weighted_metrics=weighted_metrics,
      loss_weights=loss_weights,
      sample_weight_mode=sample_weight_mode)
```
Since we have multiple losses `training_config['loss']` is a list: `[{'class_name': 'MeanSquaredError', 'config': {...}}, {'class_name': 'MeanSquaredError', 'config': {...}}]`

This case is not covered, therefore the loss configs in the list are not parsed to functions leading to errors further down the call stack.

**Comment by amahendrakar** [comment]
Was able to reproduce the issue. Please find the Gist [here](https://colab.sandbox.google.com/gist/amahendrakar/d6555cb6c92eaa51aa0e6011d1abc244/36492.ipynb). Thanks!

**Comment by ymodak** [comment]
This is fixed with latest tf nightly version ''2.2.0-dev20200218''. Thanks!
